File: erpnext/gift__management/report/soelra_report/soelra_report.py
# Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

# import frappe

# Copyright (c) 2023, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

import frappe
from datetime import datetime



def execute(filters=None):
	columns = get_columns()
	data = get_data(filters)
	return columns, data
def get_columns():
	columns = [
		  {
            'fieldname': 'name',
            'label': 'name',
            'fieldtype': 'Data',
            'options': '',
			
        },
     {
            'fieldname': 'branch',
            'label':'branch',
            'fieldtype': 'Data',
            
        },
      
        {
            'fieldname': 'date',
            'label': 'date',
            'fieldtype': 'Data',
            'options': ''
        },
        {
            'fieldname': 'remarks',
            'label': 'remarks',
            'fieldtype': 'Data',
            'options': ''
        },
            {
            'fieldname': 'company',
            'label': 'company',
            'fieldtype': 'Data',
            'options': ''
        },
   {
            'fieldname': 'vip_name',
            'label': 'vip_name',
            'fieldtype': '',
            'options': ''
        }
   ,
   {
            'fieldname': 'particulars',
            'label': 'particulars',
            'fieldtype': 'Data',
            'options': ''
        }
   ,
     
         {
            'fieldname': 'quantity',
            'label': 'quantity',
            'fieldtype': 'Data',
            'options': ''
        },
    {
            'fieldname': 'rate',
            'label': 'Rate',
            'fieldtype': 'Data',
            'options': ''
        },
		
		 {
            'fieldname': 'warehouse',
            'label': 'warehouse',
            'fieldtype': 'Data',
            'options': ''
        },
		
    {
            'fieldname': 'uom',
            'label': 'UOM',
            'fieldtype': 'Data',
            
        },
  
   
    
   
   
	]


	return columns

def get_data(filters):
    conditions = get_filters(filters)
    # data = frappe.db.get_all("Housing Application",fields=get_fields_name(), filters = conditions)
    # return data
    query = """
                     SELECT s.name ,s.branch,s.date,s.remarks,s.company,s.vip_name,sd.particulars,sd.quantity,sd.rate,sd.warehouse,sd.uom from `tabSoelra` s inner join `tabSoelra Details` sd on s.name=sd.parent where {conditions}""".format(conditions = conditions)
           
    data = frappe.db.sql(query, filters,as_dict=True)
    return data
        

def get_filters(filters):
    conditions = []
    if filters.get('application_date_time'):
      
        conditions.append("application_date_time <= %(application_date_time)s")
    if filters.get('employment_type'):
        conditions.append('employment_type = %(employment_type)s')
    if filters.get('work_station'):
        conditions.append('work_station = %(work_station)s')
    # frappe.errprint(conditions)

    return  " AND ".join(conditions) if conditions else "1=1"

File: erpnext/gift__management/report/soelra_report/soelra_report.js
// Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Soelra Report"] = {
	"filters": [
		{
            'fieldname': 'application_date_time',
            'label': 'Application Date',
            'fieldtype': 'Date',
            'options': 'Application Date',
			
        },
		
		 {
            'fieldname': 'employment_type',
            'label': 'Employment Type',
            'fieldtype': 'Select',
            'options': [
				"Civil Servant",
				"Corporation, Private and etc",
			]
        },
       
    {
            'fieldname': 'application_status',
            'label': 'Application Status',
            'fieldtype': 'Select',
            'options': [
				"pending",
				"Allotted",
				"Rejected",
				"Withdrawn",
				"Cancelled",
                "Not Eligible",
                "Resigned/Superannuated"
			]
        },
    
      {
            'fieldname': 'building_classification',
            'label': 'Eligible Building Classification',
            'fieldtype': 'Link',
            'options': 'Building Classification',
            // 'Link': 'Building Classification'
        },
	]

	
};
